pages/posts/flight-of-a-byte.tsx

```diff
@@ -0,0 +1,112 @@
+                </p>
+                <h3>IPFS to fetch the message</h3>
+                <p>
+                libp2p pubsub doesn&apos;t care about the meaning of a pubsub message. It only makes sure that all subscribers get it. Remember that the next step is to synchronize the message log, which is handled by the local message store of <code>client1</code>. After checking that it&apos;s not a repeat, the pubsub message is forwarded (code) as an EventPubSubMessage to the message store which receives it (code). This confirms that the type of pubsub message is to update the message log head, and it checks to make sure the head was not already processed. (If <code>client1</code> has already received the same message log head, maybe from another peer, then it doesn&apos;t need to do more processing. This may seem trivial, but it&apos;s important because it&apos;s efficient to exchange these small “head” messages frequently between peers.)  
+                </p>
+                <p>
+                The message store receives (code) the event for a new head which contains the <code>CID</code> of the message. The message (which may be large) is fetched with IPFS, not with libp2p pubsub (which is better for small messages). If you don&apos;t know what a <code>CID</code> is and how IPFS uses it to fetch content, then you can read all about it. Notice that using a <code>CID</code> to fetch the message (code) like “Hello” is a separate process from synchronizing heads. Maybe IPFS has already fetched the same <code>CID</code>, or maybe it can be fetched from another group member that is not <code>client2</code>. Wesh takes advantage of the efficient IPFS algorithms.
+                </p>
+                <h3>Decrypting and finishing</h3>
+                <p>
+                We&apos;re almost done! When the “Hello” message is fetched, it&apos;s placed in the local copy of the message log in the correct order. (Messages in the device&apos;s storage remain encrypted.) This internally sends the EventReplicated event which is received (code) by the Wesh message monitor. We started at the libp2p “level” which doesn&apos;t care about the meaning of pubsub messages, up through the message log level which only cares about the correct order of messages. Now at the “top” level, Wesh must use <code>client1&apos;s</code> cryptographic keys to decrypt (code) the message.
+                </p>
+                <p>
+                The decrypted message is sent internally (code) to all monitoring processes, including the GroupMessageList service (code). The <code>client1</code> application has called GroupMessageList, so the “Hello” message is sent (code) to the application which prints the message.
+                </p>
```

Comment:
Please include the code links.

Reply:
Sorry, I forgot to include the links. I just added them and I think we are good now.